source/common/upstream/cluster_manager_impl.h

```diff
@@ -111,12 +111,15 @@ class ClusterManagerInitHelper : Logger::Loggable<Logger::Id::upstream> {
 
   enum class State {
     // Initial state. During this state all static clusters are loaded. Any primary clusters
-    // are immediately initialized.
+    // are immediately begin initialization.
     Loading,
+    // In this state cluster manager waits for all primary clusters to finish initialization.
+    // This state may immediately transition to the next state iff all clusters are STATIC and
+    // without health checks enabled, since their initialization completes immediately.
```

Comment:
Technically I think it might be possible for this to happen even with health checks if the health checks immediately fail for some reason, but I'm not completely sure.

Reply:
Updated comment.